### PR TITLE
[Merged by Bors] - chore: fix casing per naming scheme

### PR DESCRIPTION
**Pull request by winstonyin**
Fix a lot of wrong casing mostly in the docstrings but also sometimes in def/theorem names. E.g. `fin 2 --> Fin 2`, `add_monoid_hom --> AddMonoidHom`

Remove `\n` from `to_additive` docstrings that were inserted by mathport.

Move files and directories with `Gcd` and `Smul` to `GCD` and `SMul`

### CODE REVIEW

File: Mathlib/Algebra/Group/Opposite.lean [hunk 1]
```diff
@@ -410,8 +410,8 @@ theorem Units.coe_opEquiv_symm {M} [Monoid M] (u : Mˣᵐᵒᵖ) :
 /-- A semigroup homomorphism `M →ₙ* N` can equivalently be viewed as a semigroup homomorphism
 `Mᵐᵒᵖ →ₙ* Nᵐᵒᵖ`. This is the action of the (fully faithful) `ᵐᵒᵖ`-functor on morphisms. -/
 @[to_additive
-      "An additive semigroup homomorphism `add_hom M N` can equivalently be viewed as an\n
-      additive semigroup homomorphism `add_hom Mᵃᵒᵖ Nᵃᵒᵖ`. This is the action of the\n
```
Comment (rwbarton) [inline Mathlib/Algebra/Group/Opposite.lean:414]: Are you sure these `\n`s should be removed?

File: Mathlib/Algebra/Group/Opposite.lean [hunk 2]
```diff
@@ -472,8 +472,8 @@ def AddHom.mulUnop {α β} [Add α] [Add β] : AddHom αᵐᵒᵖ βᵐᵒᵖ 
 /-- A monoid homomorphism `M →* N` can equivalently be viewed as a monoid homomorphism
 `Mᵐᵒᵖ →* Nᵐᵒᵖ`. This is the action of the (fully faithful) `ᵐᵒᵖ`-functor on morphisms. -/
 @[to_additive
-      "An additive monoid homomorphism `M →+ N` can equivalently be viewed as an\n
-      additive monoid homomorphism `Mᵃᵒᵖ →+ Nᵃᵒᵖ`. This is the action of the (fully faithful)\n
```
Comment (rwbarton) [inline Mathlib/Algebra/Group/Opposite.lean:476]: Ditto these

**Review by rwbarton — COMMENTED** [review]
lgtm aside from some questions about removed `\n`s from to_additive docstrings.

File: Mathlib/Algebra/Group/Opposite.lean (hunk 1 above)
Comment (winstonyin) [reply Mathlib/Algebra/Group/Opposite.lean:414]: I think so. The `\n`s were added in by mathport. See the (original mathlib3 file)[https://github.com/leanprover-community/mathlib/blob/master/src/algebra/group/opposite.lean], where new lines are inserted simply to keep the lines to within 100 characters long.

**Comment by rwbarton** [comment]
Thanks for these fixes. Could I ask that if (or when) you have any more fixes, to please make a new PR for them? Otherwise, reviewing becomes increasingly difficult.

**Comment by rwbarton** [comment]
bors merge

**Pull request closed without merging**